h/static/scripts/base/upgrade-elements.js

```diff
@@ -1,6 +1,18 @@
 'use strict';
 
 /**
+ * Mark an element as having been upgraded.
+ */
+function markReady(element) {
+  var HIDE_CLASS = 'u-js-hide-when-loading';
+  var hideOnLoad = Array.from(element.querySelectorAll('.' + HIDE_CLASS));
```

Comment:
Doesn't this break [your own rules](https://github.com/hypothesis/frontend-toolkit/blob/master/docs/css-style-guide.md#classes-used-in-code) about which classes it's ok to reference from JavaScript?

In general, I think we should probably not be using CSS classes for purposes other than styling. For stuff like this, a data attribute is probably a clearer way of doing it.